src/lib.rs

```diff
@@ -463,14 +471,185 @@ impl<'a> From<LanguageTag<String>> for LanguageTag<Cow<'a, str>> {
     }
 }
 
+#[cfg(feature = "serde")]
+impl From<SerdeLanguageTag> for LanguageTag<String> {
+    fn from(slt: SerdeLanguageTag) -> Self {
+        slt.lang_tag
+    }
+}
+
+impl<T> Into<String> for LanguageTag<T> where T: Deref<Target = str> {
+    fn into(self) -> String {
+        self.as_str().to_string()
+    }
+}
+
+impl TryFrom<String> for LanguageTag<String> {
+    type Error = LanguageTagParseError;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        LanguageTag::from_str(value.as_ref())
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<T: Serialize> Serialize for LanguageTag<T> {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        self.tag.serialize(serializer)
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de, T: Deref<Target = str> + Deserialize<'de>> Deserialize<'de> for LanguageTag<T> {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<LanguageTag<T>, D::Error> {
+        use serde::de::Error;
+
+        Self::parse(T::deserialize(deserializer)?).map_err(D::Error::custom)
+    }
+}
+
+// #[doc(cfg(feature = "serde"))] wait for this to be stable
+#[cfg(feature = "serde")]
+/// Language Tag Type that is infallable when used with `serde`. This is accomplished
+/// by having a default trait that auto-resolves to `und`. You must have crate feature `serde` enabled to use this.
+///
+/// If you are deserializing and it is an invalid tag, it will auto-resolve to the undefined(`und`) tag.
+///
+/// See [`LanguageTag`] for details.
+#[derive(Clone, Debug, Hash, Ord, PartialOrd, Eq, PartialEq)]
+pub struct SerdeLanguageTag {
```

Comment:
I believe this "infallible" language tag parsing is kind of a hack. I am not very happy to include in in `oxilangtag` as a good thing to do. I would be more in favor of leaving it out, people that have a use case for that are able to implement it again anyway. Additionaly they are likely to customize the default tag on parse error, operation that is not possible with the current API.

Reply:
Yep, feels like a hack to me as well, I will remove it. Users can implement their own if they wish.